src/Editor/Editor.cpp

```diff
@@ -225,6 +227,87 @@ void Editor::Show(float deltaTime) {
     if (currentEntity != nullptr) {
         glm::mat4 currentEntityMatrix = currentEntity->GetLocalMatrix();
 
+
+        currentEntityMatrix = currentEntity->GetLocalMatrix();
+
+        // Projection matrix.
+        glm::mat4 projectionMatrix = cameraEntity->GetComponent<Component::Lens>()->GetProjection(glm::vec2(ImGui::GetWindowSize().x, ImGui::GetWindowSize().y));
+
+        // View matrix.
+        glm::mat4 viewMatrix = glm::inverse(cameraEntity->GetModelMatrix());
+
+        // Load Vertices and Indices for the ray-triangle intersection test when the "paint mode" button is pressed.
+        if (currentEntity->loadPaintModeClicked && currentEntity->vertsLoaded == true) {
+            
+            // Delete previously loaded verts and indices.
```

Comment:
Why are we copying over the vertices and indices? Can't we just access them from the original arrays?